src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -466,15 +526,15 @@ void ProcessMarkedTypesWithInterfaces ()
 
 		void DiscoverDynamicCastableImplementationInterfaces ()
 		{
-			Debug.Assert (!_dynamicInterfaceCastableImplementationTypesDiscovered);
-
-			foreach (var assembly in _context.GetAssemblies ()) {
+			foreach (var assembly in _context.GetReferencedAssemblies ().ToArray ()) {
```

Comment:
Just to make sure I understand this - the effect of this is that we will always load full closure still. We need https://github.com/mono/linker/issues/1788 resolved to "fix" that, right?

Reply:
That's right - I added a comment here.